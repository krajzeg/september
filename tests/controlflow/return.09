void := {
	return;  # this should return Nothing
	42;      # this should not get returned
}

print("Testing empty 'return' - should yield Nothing:", void())

int := {
	return: 42 
	1 # this should not be returned
}
print("Testing literal 'return' - should yield 42:", int())

expr := |a,b| {
	return: a * a + b * b + 1
	1 # this should not be returned       
}
print("Testing expression 'return' - should yield 42:", expr(4, 5))
