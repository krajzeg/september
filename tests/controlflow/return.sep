void := {
	return;  # this should return Nothing
	42;      # this should not get returned
}

print("Testing empty 'return' - should yield Nothing:", void())
