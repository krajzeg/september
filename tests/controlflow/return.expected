Testing empty 'return' - should yield Nothing: <Nothing>
